src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.StateKind.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal partial class AnalysisState
+    {
+        /// <summary>
+        /// State kind of per-analyzer <see cref="AnalyzerStateData"/> tracking an analyzer's partial analysis state.
+        /// </summary>
+        internal enum StateKind
+        {
+            /// <summary>
+            /// Ready for processing.
```

Comment:
Going solely by the comments, the combination of "ready for processing" and "currently being processed" suggests a third state: "processed", which apparently is not necessary. Going solely by the value names, the combination of "Ready" and "InProcess" suggests that "Ready" might mean "processed". Maybe "Ready" should be changed to "ReadyToProcess", and maybe a comment explaining that there is no "processed" state is appropriate.